hazelcast/src/main/java/com/hazelcast/client/impl/connection/tcp/ExceptionCatchingConnectionProcessListener.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.impl.connection.tcp;
+
+import com.hazelcast.client.impl.management.ClientConnectionProcessListener;
+import com.hazelcast.cluster.Address;
+import com.hazelcast.logging.ILogger;
+
+import java.util.List;
+
+class ExceptionCatchingConnectionProcessListener
+        implements ClientConnectionProcessListener {
+
+    private final ClientConnectionProcessListener delegate;
+    private final ILogger logger;
+
+    ExceptionCatchingConnectionProcessListener(ClientConnectionProcessListener delegate, ILogger logger) {
+        this.delegate = delegate;
+        this.logger = logger;
+    }
+
+    private void runSafely(Runnable r) {
+        try {
+            r.run();
+        } catch (RuntimeException e) {
+            logger.severe("exception thrown by ClientConnectionProcessListener", e);
```

Comment:
This is not going to work safely when you have more than one listener. If the first listener throws, this code will not run the remaining two listeners at all

Reply:
discussed in DM: the underlying calls don't throw because they end up in `ExceptionCatchingConnectionProcessListener `